src/morphology/german/checkAdjectiveExceptions.js

```diff
@@ -0,0 +1,234 @@
+
+	for ( let i = 0; i < exceptionStems.length; i++ ) {
+		const stemPairToCheck = exceptionStems[ i ];
+
+		if ( stemPairToCheck.includes( stemmedWordToCheck ) ) {
+			return unique( [
+				stemPairToCheck[ 0 ].concat( "er" ),
+				...addAllAdjectiveSuffixes( morphologyDataAdjectives, stemPairToCheck[ 0 ].concat( "er" ) ),
+				...addComparativeSuffixes( morphologyDataAdjectives, stemPairToCheck[ 1 ] ),
+			] );
+		}
+	}
+
+	return [];
+};
+
+/**
+ * Returns forms for adjectives that get the regular suffixes on their first stem (e.g., gesund-e) and the comparative and
+ * superlative suffixes on their second stem (e.g., gesünd-er, gesünd-est).
+ *
+ * @param {Object}  morphologyDataAdjectives The German morphology data for nouns.
+ * @param {string}  stemmedWordToCheck       The stem to check.
+ *
+ * @returns {string[]} The created adjective forms.
+ */
+const secondStemCompSup = function( morphologyDataAdjectives, stemmedWordToCheck ) {
+	const exceptionStems = morphologyDataAdjectives.secondStemCompSup;
+
+	for ( let i = 0; i < exceptionStems.length; i++ ) {
+		const stemPairToCheck = exceptionStems[ i ];
+
+		if ( stemPairToCheck.includes( stemmedWordToCheck ) ) {
+			return unique( [
+				...addRegularSuffixes( morphologyDataAdjectives, stemPairToCheck[ 0 ] ),
+				...addComparativeSuffixes( morphologyDataAdjectives, stemPairToCheck[ 1 ] ),
+				...addSuperlativeSuffixes( morphologyDataAdjectives, stemPairToCheck[ 1 ] ),
+			] );
+		}
+	}
+
+	return [];
+};
+
+/**
+ * Returns forms for adjectives that get all suffixes on the first stem (e.g., blass-e, blass-er, blass-est)
+ * and only the comparative and superlative suffixes on the second (bläss-er, bläss-est).
+ *
+ * @param {Object}  morphologyDataAdjectives The German morphology data for nouns.
+ * @param {string}  stemmedWordToCheck       The stem to check.
+ *
+ * @returns {string[]} The created adjective forms.
+ */
+const bothStemsComSup = function( morphologyDataAdjectives, stemmedWordToCheck ) {
+	const exceptionStems = morphologyDataAdjectives.bothStemsCompSup;
+
+	for ( let i = 0; i < exceptionStems.length; i++ ) {
+		const stemPairToCheck = exceptionStems[ i ];
+
+		if ( stemPairToCheck.includes( stemmedWordToCheck ) ) {
+			return unique( [
+				...addAllAdjectiveSuffixes( morphologyDataAdjectives, stemPairToCheck[ 0 ] ),
+				...addComparativeSuffixes( morphologyDataAdjectives, stemPairToCheck[ 1 ] ),
+				...addSuperlativeSuffixes( morphologyDataAdjectives, stemPairToCheck[ 1 ] ),
+			] );
+		}
+	}
+
+	return [];
+};
+
+/**
+ * Checks whether a give stem stem falls into any of the adjective exception categories.
+ *
+ * @param {Object}  morphologyDataAdjectives The German morphology data for nouns.
+ * @param {string}  stemmedWordToCheck       The stem to check.
+ *
+ * @returns {string[]} The created adjective forms.
+ */
+export function checkAdjectiveExceptions( morphologyDataAdjectives, stemmedWordToCheck ) {
```

Comment:
Same for this function:
Change this to, for example, `generateAdjectiveExceptionForms`. To indicate that it not only checks, but also generates the exception forms.